allure-testng/src/main/java/io/qameta/allure/testng/AllureTestNg.java

```diff
@@ -769,6 +776,27 @@ private Current refreshContext() {
         return currentTestResult.get();
     }
 
+    @SuppressWarnings({"Indentation", "PMD.ExcessiveMethodLength"})
+    @Override
+    public void beforeDataProviderExecution(final IDataProviderMethod dpMethod,
+                                            final ITestNGMethod method,
+                                            final ITestContext context) {
+        startBefore(getUniqueUuid(context), method);
+    }
+
+    @SuppressWarnings({"Indentation", "PMD.ExcessiveMethodLength"})
+    @Override
+    public void afterDataProviderExecution(final IDataProviderMethod dpMethod,
+                                           final ITestNGMethod method,
+                                           final ITestContext context) {
+        final String executableUuid = currentExecutable.get();
+        getLifecycle().updateFixture(executableUuid, result -> result.setStatus(Status.PASSED));
```

Comment:
Shouldn't we update the status according to data provider execution state? As we can still fail in DP, and it'll affect the overall test's execution process.

Reply:
If the DP fails, it doesn't get to the afterDataProviderExecution method implementation at all. So I was assuming, that if this method is reached, then it has to be a success. 
The test itself is not run at all in the case of the DP failure.

Reply:
That's weird and seems like a bug (probably in TestNG). As if DP fails, we may still want to perform a cleanup of created entities. Skipping the test sounds reasonable. But hooks designed as configuration methods, e.g. if a test fails, 'after' hook is still called.

Reply:
I was trying to attach to some listener method on DP failure, but could not find anything useful. Even in ITestListener#onTestSkipped, exposed ITestResult object's getSkipCausedBy() method returns empty list. 
So, seems like DP is not implemented perfectly... IDataProviderMethod is not related to ITestNGMethod in any way.
So, do You like me to change something, or we can leave it as it is?

Reply:
Actually, I was thinking about one change in how to make the result of the DP correct on failure. 
How about setting the status of the FixtureResult to broken by default in before DP method:

```
@Override
    public void beforeDataProviderExecution(final IDataProviderMethod dpMethod,
                                            final ITestNGMethod method,
                                            final ITestContext context) {
        startBefore(getUniqueUuid(context), method);
        final String executableUuid = currentExecutable.get();
        getLifecycle().updateFixture(executableUuid, result -> result.setStatus(Status.BROKEN));
    }
```

and then if the DP is executed correctly, anyway we update the test fixture to Success 
```
@Override
    public void afterDataProviderExecution(final IDataProviderMethod dpMethod,
                                           final ITestNGMethod method,
                                           final ITestContext context) {
        final String executableUuid = currentExecutable.get();
        getLifecycle().updateFixture(executableUuid, result -> result.setStatus(Status.PASSED));
        getLifecycle().stopFixture(executableUuid);
        final String containerUuid = currentTestContainer.get();
        currentTestContainer.remove();
        validateContainerExists(getQualifiedName(method), containerUuid);
    }
```
this way we can get broken DP in the report in case of the DP failure.